src/eterna/pose2D/Pose2D.ts

```diff
@@ -317,39 +349,61 @@ export default class Pose2D extends ContainerObject implements Updatable {
         return this._zoomLevel;
     }
 
+    /**
+     * @param animate should the zoom happen in an animation over time, instead of instantaniously
+     * @param center should the camera move to the center of the screen
+     */
     public setZoomLevel(zoomLevel: number, animate: boolean = true, center: boolean = false): void {
-        if ((this._zoomLevel !== zoomLevel || center) && animate) {
-            if (this._zoomLevel === zoomLevel && center) {
-                if (Math.abs(this._width / 2 - this._offX) + Math.abs(this._height / 2 - this._offY) < 50) {
+        const needsToZoom = this._zoomLevel !== zoomLevel;
+        if ((needsToZoom || center) && animate) {
+            if (!needsToZoom && center) {
+                // TODO Guy: hypot is expensive and I don't even need the sqrt
+                const distanceFromCenter = Math.hypot(
+                    this._width / 2 - this._offset.x,
+                    this._height / 2 - this._offset.y
+                );
+                if (distanceFromCenter < 50) {
                     return;
                 }
             }
 
-            this._startOffsetX = this._offX;
-            this._startOffsetY = this._offY;
-
-            let scaler = 1;
-            if (zoomLevel > this._zoomLevel) {
-                scaler = Pose2D.ZOOM_SPACINGS[zoomLevel] / Pose2D.ZOOM_SPACINGS[this._zoomLevel];
-            }
+            this._startOffset = this._offset.clone();
 
-            if (!this._offsetTranslating && !center) {
-                this._endOffsetX = scaler * (this._offX - this._width / 2) + this._width / 2;
-                this._endOffsetY = scaler * (this._offY - this._height / 2) + this._height / 2;
-            } else if (this._offsetTranslating) {
-                this._endOffsetX = scaler * (this._endOffsetX - this._width / 2) + this._width / 2;
-                this._endOffsetY = scaler * (this._endOffsetY - this._height / 2) + this._height / 2;
+            if (center) {
+                this._endOffset = new Point(this._width / 2, this._height / 2);
             } else {
-                this._endOffsetX = this._width / 2;
-                this._endOffsetY = this._height / 2;
+                // TODO: should this be different? I need to respect if the mosue moved
+                // also what if the user presses the button?
+                // also what if the user drags in the middle of the zoom?
```

Comment:
Dragging in the middle of the zoom is broken in live, and I don't see myself fixing it.